pkg/util/winutil/process.go

```diff
@@ -88,9 +90,12 @@ func ReadProcessMemory(h windows.Handle, from, to uintptr, count uint32) (bytesR
 		uintptr(unsafe.Pointer(&bytes)))
 
 	if r == 0 {
-		err = windows.GetLastError()
-		fmt.Printf("Error %v %v\n", err, e)
-		return
+		if e == windows.ERROR_ACCESS_DENIED {
+			log.Debugf("Access denied error getting process memory ")
+		} else {
+			log.Warnf("Unexpected error getting process memory for handle (h) %v (err) %v", h, e)
```

Comment:
Are we losing any valuable data now by not including the result of `windows.GetLastError()`? If so, can we include that data in the output? If we're not losing any data, can we rename the `e` var to `err` for consistency with other code?

Reply:
That was part of the problem.  GetLastError() is getting cleared in the syscall, and is therefore returning null, so it's useless.
The reason I used another variable (`e`), is that when I used `err` in the assignment (`:=`) above, it hid the return variable of the same name and we ended up returning `nil` for the error.

Reply:
Not a blocker but I think that the named variables in the method signature are really not providing any value here since we have a mixture of explicit returns (`return 0, e`) and implicit ones (`return` at the bottom) plus we also have the variable hiding issue that you mention. I think it may be valuable to just make this whole method use explicit returns but up to you how you want to handle this.
```
func ReadProcessMemory(h windows.Handle, from, to uintptr, count uint32) (uint64, error) {
```

At the very least we should consistently either use implicit or explicit returns.